vortex-array/src/array/display/mod.rs

```diff
@@ -0,0 +1,287 @@
+    /// Only the logical values of the array: `[0i16, 1i16, 2i16, 3i16, 4i16]`.
+    ///
+    /// ```
+    /// # use vortex_array::display::DisplayOptions;
+    /// # use vortex_array::IntoArray;
+    /// # use vortex_buffer::buffer;
+    /// let array = buffer![0_i16, 1, 2, 3, 4].into_array();
+    /// let opts = DisplayOptions::default();
+    /// assert_eq!(
+    ///     format!("{}", array.display_as(&opts)),
+    ///     "[0i16, 1i16, 2i16, 3i16, 4i16]",
+    /// );
+    /// assert_eq!(
+    ///     format!("{}", array.display_as(&opts)),
+    ///     format!("{}", array.display()),
+    /// );
+    /// ```
+    CommaSeparatedScalars { space_after_comma: bool },
+    /// The tree of encodings and all metadata but no values.
+    ///
+    /// ```
+    /// # use vortex_array::display::DisplayOptions;
+    /// # use vortex_array::IntoArray;
+    /// # use vortex_buffer::buffer;
+    /// let array = buffer![0_i16, 1, 2, 3, 4].into_array();
+    /// let opts = DisplayOptions::TreeDisplay;
+    /// let expected = "root: vortex.primitive(i16, len=5) nbytes=10 B (100.00%)
+    ///   metadata: EmptyMetadata
+    ///   buffer (align=2): 10 B (100.00%)
+    /// ";
+    /// assert_eq!(format!("{}", array.display_as(&opts)), expected);
+    /// ```
+    TreeDisplay,
+}
+
+impl Default for DisplayOptions {
+    fn default() -> Self {
+        Self::CommaSeparatedScalars {
+            space_after_comma: true,
+        }
+    }
+}
+
+/// A shim used to display an array as specified in the options.
+///
+/// See also:
+/// [Array::display](http://localhost:8000/doc/vortex_array/trait.Array.html#method.display),
+/// [Array::display_as](http://localhost:8000/doc/vortex_array/trait.Array.html#method.display_as),
+/// [DisplayArray], and [DisplayOptions].
+pub struct DisplayArrayAs<'a>(pub &'a dyn Array, pub &'a DisplayOptions);
+
+impl Display for DisplayArrayAs<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt_as(f, self.1)
+    }
+}
+
+/// Display the encoding and limited metadata of this array.
+///
+/// # Examples
+/// ```
+/// # use vortex_array::IntoArray;
+/// # use vortex_buffer::buffer;
+/// let array = buffer![0_i16, 1, 2, 3, 4].into_array();
+/// assert_eq!(
+///     format!("{}", array),
+///     "vortex.primitive(i16, len=5)",
+/// );
+/// ```
+impl Display for dyn Array + '_ {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.fmt_as(f, &DisplayOptions::MetadataOnly)
+    }
+}
+
+impl dyn Array + '_ {
+    /// Display the logical values of the array.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use vortex_array::IntoArray;
+    /// # use vortex_buffer::buffer;
+    /// let array = buffer![0_i16, 1, 2, 3, 4].into_array();
+    /// assert_eq!(
+    ///     format!("{}", array.display()),
+    ///     "[0i16, 1i16, 2i16, 3i16, 4i16]",
+    /// )
+    /// ```
+    pub fn display(&self) -> DisplayArray {
+        DisplayArray(self)
```

Comment:
Why is this called `display`? When it's different from `impl Display`? It's quite confusing. I would just use display_as always